tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/object/test_objects.bal

```diff
@@ -466,6 +466,20 @@ function testSubTypingForDifferentOrgNameAndVersionWithAllPublicFields() {
     assertTrue((<any> ob3) is foo2:ObjectWithPublicFieldsAndMethods);
 }
 
+readonly client class FooClass {
+    *foo2:FooObj;
```

Comment:
Missed this earlier. But shall we add a test where the included object is an intersection with a class too?

Reply:
e.g., `x:T` where `T` is `readonly & Class`.

Reply:
Fixed via https://github.com/ballerina-platform/ballerina-lang/pull/40386/commits/28a13d2b701110fcf68600a89f504a54ea56f252